x-pack/plugins/screenshotting/server/plugin.ts

```diff
@@ -65,9 +66,9 @@ export class ScreenshottingPlugin implements Plugin<void, ScreenshottingStart, S
 
         return new HeadlessChromiumDriverFactory(this.screenshotMode, config, logger, binaryPath);
       } catch (error) {
+        this.error = error;
```

Comment:
There is an easier and cleaner way to achieve the same results without introducing new property.

```typescript
    this.browserDriverFactory = (async () => {
      const paths = new ChromiumArchivePaths();
      const logger = this.logger.get('chromium');
      const [config, binaryPath] = await Promise.all([
        createConfig(this.logger, this.config),
        install(paths, logger),
      ]);

      return new HeadlessChromiumDriverFactory(this.screenshotMode, config, logger, binaryPath);
    })();

    this.browserDriverFactory.catch(() => {
      this.logger.error('Error in screenshotting setup, it may not function properly.');
    });
```

Reply:
This change will fix unhandled promise rejection, and the error will be stored in the promise instance.